content/blog/esc-db-secrets-rotation-launch/index.md

```diff
@@ -0,0 +1,193 @@
+date: 2025-04-30 # Adjust T00:00:00-00:00 as needed for timezone/exact time
+allow_long_title: true
+meta_desc: "Pulumi ESC now automates the rotation of database credentials for PostgreSQL, MySQL (on AWS), and Snowflake, enhancing security and reducing operational burden."
+meta_image: meta.png
+authors:
+- sean-yeh
+- claire-gaestel
+- arun-loganathan
+tags:
+- esc
+- secrets
+- features
+- rotation
+- database
+- security
+---
+
+Securing access to critical data stores is paramount in today's cloud-native world. Yet, managing database credentials often involves static, long-lived passwords – a significant security blind spot. These static secrets, frequently embedded in application configurations or accessible to multiple team members, represent a prime target for attackers. Manually rotating these credentials is a cumbersome, error-prone task that’s often neglected, leaving databases vulnerable for extended periods. Building on our commitment to robust secrets management, we are excited to launch **Automated Database Credential Rotation** for **PostgreSQL, MySQL, and Snowflake** in [Pulumi ESC!](/product/esc)
+
+<!--more-->
+
+## The Persistent Challenge of Static Database Secrets
+
+Relying on static database credentials introduces substantial risks, which automated rotation helps mitigate:
+
+*   **Security Vulnerabilities & Exposure:** Static credentials, if compromised through leaks, phishing, or unauthorized access, provide long-term access to attackers. Automated rotation significantly shrinks this window of opportunity.
+*   **Operational & Compliance Burdens:** Manually rotating database credentials is complex, error-prone, and requires careful coordination to avoid downtime. It also makes demonstrating compliance with regulations like SOC 2, GDPR, or HIPAA (which often mandate rotation) difficult and hard to audit consistently.
+*   **Network Complexity:** Many databases reside within private networks (like AWS VPCs) for security. Rotating credentials for these databases typically requires bridging this network gap securely, often involving mechanisms like proxies, bastion hosts, or specific agents. This adds significant setup and maintenance overhead for manual or homegrown solutions.
+*   **Tooling and Ecosystem Compatibility:** Many standard database administration GUIs, Business Intelligence platforms, ETL tools, and custom scripts were built primarily for username/password authentication. While support for cloud provider-specific methods like AWS IAM database authentication is growing, it's not universal. Older versions, generic ODBC/JDBC connectors, or specific tools may lack native support or require complex configuration, making username/password the more straightforward or only viable connection method for these essential parts of the data ecosystem, reinforcing the need for automated password rotation.
+
+Pulumi ESC addresses these challenges by automating the rotation process, including databases in private networks.
```

Comment:
Should we be explict that we support AWS for now, and have link to premade Azure and GCP items for upvoting?
Perhaps Oracle and CloudFlare too.